internal/resources/diffValidation/resource_nsxt_lb_profile.go.go

```diff
@@ -0,0 +1,131 @@
+//  (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package diffvalidation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	serviceTypes    = "service_type"
+	httpProfile     = "http_profile"
+	tcpProfile      = "tcp_profile"
+	udpProfile      = "udp_profile"
+	cookieProfile   = "cookie_profile"
+	sourceIpProfile = "sourceip_profile"
+	genericProfile  = "generic_profile"
+	clientProfile   = "client_profile"
+	serverProfile   = "server_profile"
+
+	LBHttpProfile                = "LBHttpProfile"
+	LBFastTcpProfile             = "LBFastTcpProfile"
+	LBFastUdpProfile             = "LBFastUdpProfile"
+	LBCookiePersistenceProfile   = "LBCookiePersistenceProfile"
+	LBSourceIpPersistenceProfile = "LBSourceIpPersistenceProfile"
+	LBGenericPersistenceProfile  = "LBGenericPersistenceProfile"
+	LBClientSslProfile           = "LBClientSslProfile"
+	LBServerSslProfile           = "LBServerSslProfile"
+)
+
+type LoadBalancerProfile struct {
+	diff *schema.ResourceDiff
+}
+
+func NewLoadBalancerProfileValidate(diff *schema.ResourceDiff) *LoadBalancerProfile {
+	return &LoadBalancerProfile{
+		diff: diff,
+	}
+}
+
+func (l *LoadBalancerProfile) DiffValidate() error {
+	err := l.validateProfile()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *LoadBalancerProfile) validateProfile() error {
+	if l.diff.HasChange(serviceTypes) {
+		service := l.diff.Get(serviceTypes)
+		profileType := l.diff.Get(httpProfile)
+		if service == LBHttpProfile {
+			if len((profileType).([]interface{})) == 0 {
+				return fmt.Errorf("please provide http_profile configurations for serviceType LBHttpProfile")
+			}
+		}
+	}
+
+	if l.diff.HasChange(serviceTypes) {
```

Comment:
`func (l *LoadBalancerMonitor) validateMonitor(monitor_type string, service_type string ) error {
    if l.diff.HasChange(serviceType) {
        service := l.diff.Get(serviceType)
        monitorType := l.diff.Get(monitor_type)
        if service == service_type {
            if len((monitorType).([]interface{})) == 0 {
                return fmt.Errorf("please provide "+ monitor_type +"configurations for Type LBTcpMonitorProfile")
            }
        }
    }
    return nil
}`

Reply:
updated the fixes